load-gen/src/main/java/io/smartcat/data/loader/api/Terminateable.java

```diff
@@ -0,0 +1,13 @@
+package io.smartcat.data.loader.api;
+
+/**
+ * Has ability to terminate itself.
+ */
+public interface Terminateable {
```

Comment:
To me, it seems that I was wrong in first place, that lifetime of those components if not responsibility of LoadGenerator, it is out of its scope.